themes/pzq/pzq.theme.sh

```diff
@@ -0,0 +1,170 @@
+D_BRANCH_COLOR="${cyan}"
+D_CHANGES_COLOR="${white}"
+D_TIME_COLOR="${white}"
+D_DOLLOR_COLOR="${bold_red}"
+D_CMDFAIL_COLOR="${red}"
+D_VIMSHELL_COLOR="${cyan}"
+
+# ------------------------------------------------------------------ FUNCTIONS
+case $TERM in
+  xterm*)
+      TITLEBAR="\033]0;\w\007"
+      ;;
+  *)
+      TITLEBAR=""
+      ;;
+esac
+
+is_vim_shell() {
+  if [ ! -z "$VIMRUNTIME" ];
+  then
+    echo "${D_INTERMEDIATE_COLOR}on ${D_VIMSHELL_COLOR}\
+vim shell${D_DEFAULT_COLOR} "
+  fi
+}
+
+mitsuhikos_lastcommandfailed() {
+  code=$?
+  if [ $code != 0 ];
+  then
+    echo " ${D_DEFAULT_COLOR}C:${D_CMDFAIL_COLOR}\
+$code ${D_DEFAULT_COLOR}"
+  fi
+}
+
+# vcprompt for scm instead of oh-my-bash default
+demula_vcprompt() {
+  if [ ! -z "$VCPROMPT_EXECUTABLE" ];
+  then
+    local D_VCPROMPT_FORMAT="on ${D_SCM_COLOR}%s${D_INTERMEDIATE_COLOR}:\
+${D_BRANCH_COLOR}%b %r ${D_CHANGES_COLOR}%m%u ${D_DEFAULT_COLOR}"
+    $VCPROMPT_EXECUTABLE -f "$D_VCPROMPT_FORMAT"
+  fi
+}
+
+# checks if the plugin is installed before calling battery_charge
+safe_battery_charge() {
+  if [ -e "${OSH}/plugins/battery/battery.plugin.sh" ];
+  then
+    battery_charge
+  fi
+}
+
+prompt_git() {
+	local branchName='';
+
+	# Check if the current directory is in a Git repository.
+	if [ $(git rev-parse --is-inside-work-tree &>/dev/null; echo "${?}") == '0' ]; then
+		# Get the short symbolic ref.
+		# If HEAD isn’t a symbolic ref, get the short SHA for the latest commit
+		# Otherwise, just give up.
+		branchName="$(git symbolic-ref --quiet --short HEAD 2> /dev/null || \
+			git rev-parse --short HEAD 2> /dev/null || \
+			echo '(unknown)')";
+
+		echo -e "${D_GIT_DEFAULT_COLOR}on ${D_GIT_BRANCH_COLOR}${branchName} ";
+	else
+		return;
+	fi;
+}
+
+limited_pwd() {
+  # Max length of PWD to display
+  MAX_PWD_LENGTH=20
+
+  # Replace $HOME with ~ if possible 
+  RELATIVE_PWD=${PWD/#$HOME/\~}
+
+  local offset=$((${#RELATIVE_PWD}-$MAX_PWD_LENGTH))
+
+  if [ $offset -gt "0" ]
+  then
+      local truncated_symbol="..."
+      TRUNCATED_PWD=${RELATIVE_PWD:$offset:$MAX_PWD_LENGTH}
+      echo -e "${truncated_symbol}/${TRUNCATED_PWD#*/}"
+  else
+      echo -e "${RELATIVE_PWD}"
+  fi
+}
+
+prompt_python() {
+  # for virtualenv
+  if [[ $VIRTUAL_ENV ]]; then
+    virtualenv=`basename "$VIRTUAL_ENV"`
+    VIRTUALENV_THEME_PROMPT_PREFIX='('
+    VIRTUALENV_THEME_PROMPT_SUFFIX=')'
+    echo -e "${VIRTUALENV_THEME_PROMPT_PREFIX}$virtualenv${VIRTUALENV_THEME_PROMPT_SUFFIX}"
+  fi
+
+  # for condaenv
+  if [[ $CONDA_DEFAULT_ENV ]]; then
+    CONDAENV_THEME_PROMPT_PREFIX='('
+    CONDAENV_THEME_PROMPT_SUFFIX=')'
+    echo -e "${CONDAENV_THEME_PROMPT_PREFIX}${CONDA_DEFAULT_ENV}${CONDAENV_THEME_PROMPT_SUFFIX}"
+  fi
+}
+
+# -------------------------------------------------------------- PROMPT OUTPUT
+prompt() {
+  local LAST_COMMAND_FAILED=$(mitsuhikos_lastcommandfailed)
+  local SAVE_CURSOR='\033[s'
+  local RESTORE_CURSOR='\033[u'
+  local MOVE_CURSOR_RIGHTMOST='\033[500C'
```

Comment:
500 might not be sufficient for this 4K era...

```suggestion
  local MOVE_CURSOR_RIGHTMOST='\033['${COLUMNS:-9999}'C'
```